src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -90,55 +148,131 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
         );
     }
 
+    readonly xProfileOptions = remoteData<{label:string, value:string}[]> ({
+        await: ()=>[
+            this.props.store.coexpressionTabMolecularProfiles,
+            this.isSelectedGeneticEntityAGeneSet
+        ],
+        invoke: () => {
+            let filteredProfiles: MolecularProfile[] = [];
+            if (this.isSelectedGeneticEntityAGeneSet.result!) {
+                filteredProfiles = getGenesetProfiles(this.props.store.coexpressionTabMolecularProfiles.result!);
+            } else {
+                filteredProfiles = filterAndSortProfiles(this.props.store.coexpressionTabMolecularProfiles.result!);
+            }
+            return Promise.resolve(filteredProfiles.map(profile=>{
```

Comment:
this part is duplicated between x and y profile options, can you pull it out into a helper method in CoExpressionUtils, e.g. `export function getProfileOptions(profiles:MolecularProfile[], profileToProfiledSampleCount:{[id:string]:number}){`